### DISCUSSION TITLE
Expand and collapse rows with drag-and-drop

### DISCUSSION THREAD

**Discussion by TomPast** [TanStack/table#3230] [Q&A]
Hey, is there any way I can use expand and collapse rows with the drag and drop functionality. For example, i would like to drag and drop a complete row with his subrows under another row.

Expandable and collapse rows : https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/row-dnd
Drag and Drop rows : https://codesandbox.io/s/jolly-payne-dxs1d?fontsize=14&hidenavigation=1&theme=dark

**Comment by gregoryforel** [comment]
Hi @TomPast ,

I haven't tried, but if you look at the source from `row-dnd` and get some inspiration from how [this component](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeList/LocalReordering/React/Light/) does it, you might be able to.

It looks like they consider 2 different drop areas. The first would be to drop directly _on_ a node. In that case, the dropped node becomes a child. The second one is to drop _between_ 2 nodes, and this re-orders the nodes.

@tannerlinsley : hello Tanner! Do you foresee any reason why this behaviour could not be implemented in `react-table`?
